packages/amplify-cli/src/__tests__/init-steps/s0-analyzeProject.test.ts

```diff
@@ -0,0 +1,42 @@
+import { $TSContext } from 'amplify-cli-core';
+import { analyzeProject } from '../../init-steps/s0-analyzeProject';
+import { constructMockPluginPlatform } from '../extensions/amplify-helpers/mock-plugin-platform';
+import { Input } from '../../domain/input';
+import { constructContext } from '../../context-manager';
+
+describe('analyzeProject', () => {
+  it('recognizes the default editor', async () => {
+    const mockPluginPlatform = constructMockPluginPlatform();
+    const mockProcessArgv = [
+      '/Users/userName/.nvm/versions/node/v12.16.1/bin/node',
+      '/Users/userName/.nvm/versions/node/v12.16.1/bin/amplify',
+      'init',
+      '-y',
+    ];
+    const mockInput = new Input(mockProcessArgv);
+    const mockContext = constructContext(mockPluginPlatform, mockInput) as unknown as $TSContext;
+    const frontendPlugins = [
+      {
+        name: 'amplify-frontend-javascript',
+        pluginType: 'frontend',
+        pluginName: 'javascript',
+        directory: 'amplify-frontend-javascript',
+      },
+      {
+        name: 'amplify-frontend-flutter',
+        pluginType: 'frontend',
+        pluginName: 'flutter',
+        directory: 'amplify-frontend-flutter',
+      },
+    ];
+    mockContext.exeInfo = {
+      inputParams: {},
+      localEnvInfo: {
+        defaultEditor: 'Visual Studio Code',
```

Comment:
Could you please use a defaultEditor value from this list https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli/src/extensions/amplify-helpers/editor-selection.ts#L5-L42 . ( please use the value "vscode" instead of the name "Visual Studio Code" in this test)

Reply:
Sure! I just want to let you know that I modeled my `mockContext.exeInfo` object by capturing at the attribute during `amplify init -y` and I found the `defaultEditor` value to be `Visual Studio Code`, which is why I used the same. 

![image](https://user-images.githubusercontent.com/40818234/155399713-7213c694-2581-4e69-9118-0a6e6eb9e681.png)

If you'd still like, please let me know and I can change it to `vscode`.

Reply:
Thanks! 
it could be that on Windows the executable is named "Visual Studio Code". no need to change then.